bson/bsoncodec/registry.go

```diff
@@ -359,14 +375,21 @@ func (r *Registry) lookupTypeEncoder(t reflect.Type) (ValueEncoder, bool) {
 	return enc, found
 }
 
-func (r *Registry) lookupInterfaceEncoder(t reflect.Type) (ValueEncoder, bool) {
+func (r *Registry) lookupInterfaceEncoder(t reflect.Type, allowAddr bool) (ValueEncoder, bool) {
 	if t == nil {
 		return nil, false
 	}
 	for _, ienc := range r.interfaceEncoders {
-		if t.Implements(ienc.i) || reflect.PtrTo(t).Implements(ienc.i) {
+		if t.Implements(ienc.i) {
 			return ienc.ve, true
 		}
+		if allowAddr && t.Kind() != reflect.Ptr && reflect.PtrTo(t).Implements(ienc.i) {
+			defaultEnc, found := r.lookupInterfaceEncoder(t, false)
```

Comment:
Is this nested call to `lookupInterfaceEncoder` needed for the case where `*T` implements this the interface we're checking for but `T` implements some other interface that's later in the further ahead in the `interfaceEncoders` list?

Reply:
I'm fine with this. Let's add a comment to explain it, though.